### PR TITLE
feat(autoware_behavior_path_dynamic_obstacle_avoidance): expand drivable area

### PR DESCRIPTION
**Pull request by beyzanurkaya**
## Description
fixes: https://github.com/autowarefoundation/autoware.universe/issues/7690

## Related links

**Parent Issue:** https://github.com/autowarefoundation/autoware.universe/issues/7634
Must merged: https://github.com/autowarefoundation/autoware_launch/pull/1133

## How was this PR tested?

https://github.com/user-attachments/assets/48275d67-a09d-43e7-9597-c870d1ee7463

## Notes for reviewers

None.

## Interface changes

| Parameter Name   |    Type     | Default Value     | Description       |
|:-------------------------|:------------  |:--------------------- |:------------------|
| `expand_drivable_area`     | `bool`    | `false`  | expand drivable area while avoiding |

## Effects on system behavior

None.

### CODE REVIEW

**Comment by beyzanurkaya** [comment]
I achieved to expand the drivable area, but the expansion ends the moment the EGO passes the object.

https://github.com/user-attachments/assets/2a59e010-c7e5-431a-8a38-65b81b57fd5a

**Comment by beyzanurkaya** [comment]
Last Test Result:


https://github.com/user-attachments/assets/48275d67-a09d-43e7-9597-c870d1ee7463

**Comment by yuki-takagi-66** [comment]
@beyzanurkaya 
Thank you for the improved PR.
We are welcome for this PR.  However, as the Tier4 side, we would like to disable this feature because the current behavior is not safe enough for use on public loads.
Therefore, can I ask you to prepare the flag parameter for the new feature and set it as disabled?

**Comment by beyzanurkaya** [comment]
@yuki-takagi-66 -san I add a flag for the expansion feature.

File: planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/src/scene.cpp [hunk 1]
```diff
@@ -438,21 +498,10 @@ BehaviorModuleOutput DynamicObstacleAvoidanceModule::plan()
       appendExtractedPolygonMarker(debug_marker_, obstacle_poly.value(), object.pose.position.z);
     }
   }
-
-  DrivableAreaInfo current_drivable_area_info;
-  current_drivable_area_info.drivable_lanes =
```
Comment (yuki-takagi-66) [inline planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/src/scene.cpp:443]: I prefer to maintain this code, in order to keep conventional behavior.

File: planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/include/autoware/behavior_path_dynamic_obstacle_avoidance_module/scene.hpp [hunk 2]
```diff
@@ -350,6 +359,7 @@ class DynamicObstacleAvoidanceModule : public SceneModuleInterface
     parameters_ = std::any_cast<std::shared_ptr<DynamicAvoidanceParameters>>(parameters);
   }
 
+  bool arrived_path_end_{false};
```
Comment (yuki-takagi-66) [inline planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/include/autoware/behavior_path_dynamic_obstacle_avoidance_module/scene.hpp:362]: If we can separate this feature from the drivable area expansion, I suggest to make separate PR.

**Review by yuki-takagi-66 — COMMENTED** [review]
Thank you for your response. I made some comments, could you take a look?

File: planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/include/autoware/behavior_path_dynamic_obstacle_avoidance_module/scene.hpp (hunk 2 above)
Comment (beyzanurkaya) [reply planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/include/autoware/behavior_path_dynamic_obstacle_avoidance_module/scene.hpp:362]: I had to change the contents of canTransitSuccessState so that the expansion would stop at the right time. But I can create a PR for that part if you want.

```
const auto & data = getPreviousModuleOutput();
  const auto idx = planner_data_->findEgoIndex(data.reference_path.points);
  if (idx == data.reference_path.points.size() - 1) {
    arrived_path_end_ = true;
  }
  constexpr double THRESHOLD = 2.0;
  const auto is_further_than_threshold =
    autoware::universe_utils::calcDistance2d(
      getEgoPose(), autoware::universe_utils::getPose(data.reference_path.points.back())) >
    THRESHOLD;
  if (is_further_than_threshold && arrived_path_end_) {
    RCLCPP_WARN(getLogger(), "Reach path end point. Exit.");
    return true;
  }
```

**Comment by beyzanurkaya** [comment]
Thanks for your review @yuki-takagi-66 -san. I edited the code according to your comments.

File: planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/config/dynamic_obstacle_avoidance.param.yaml [hunk 3]
```diff
@@ -3,6 +3,7 @@
     dynamic_avoidance:
       common:
         enable_debug_info: false
+        use_lane_type: "opposite_direction_lane"
```
Comment (yuki-takagi-66) [inline planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/config/dynamic_obstacle_avoidance.param.yaml:6]: This line is not referenced.
please delete this line or declare_parameter

File: planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/include/autoware/behavior_path_dynamic_obstacle_avoidance_module/scene.hpp [hunk 4]
```diff
@@ -169,6 +173,11 @@ struct LatFeasiblePaths
 class DynamicObstacleAvoidanceModule : public SceneModuleInterface
 {
 public:
+  static constexpr const char * logger_namespace =
+    "planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner.dynamic_"
+    "obstacle_"
+    "avoidance";
```
Comment (yuki-takagi-66) [inline planning/behavior_path_planner/autoware_behavior_path_dynamic_obstacle_avoidance_module/include/autoware/behavior_path_dynamic_obstacle_avoidance_module/scene.hpp:179]: please realign the char

**Comment by yuki-takagi-66** [comment]
> @yuki-takagi-66 -san, I made the changes you wanted.

Thank you. I'll review

**Review by yuki-takagi-66 — APPROVED** [review]
Thank you. LGTM
